pkg/v7/chartconfig/spec.go

```diff
@@ -0,0 +1,21 @@
+package chartconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+
+	"github.com/giantswarm/cluster-operator/pkg/v7/key"
+)
+
+type Interface interface {
+	GetDesiredState(ctx context.Context, clusterConfig ClusterConfig, providerChartSpecs []key.ChartSpec) ([]*v1alpha1.ChartConfig, error)
+}
```

Comment:
For now just has the desired state. Eventually will be similar to the configmap interface.

https://github.com/giantswarm/cluster-operator/blob/ac33a86f9b08216860a2d84be7af831a52612c7d/pkg/v7/configmap/spec.go#L10-L18